packages/graphql/src/translate/translate-read.ts

```diff
@@ -29,17 +29,33 @@ import translateTopLevelMatch from "./translate-top-level-match";
 function translateRead({ node, context }: { context: Context; node: Node }): [string, any] {
     const { resolveTree } = context;
     const { fieldsByTypeName } = resolveTree;
-    const optionsInput = resolveTree.args.options as GraphQLOptionsArg;
     const varName = "this";
 
     let matchAndWhereStr = "";
     let authStr = "";
-    let offsetStr = "";
-    let limitStr = "";
-    let sortStr = "";
     let projAuth = "";
     let projStr = "";
 
+    const optionsInput = resolveTree.args.options as GraphQLOptionsArg;
+    let limitStr = "";
+    let offsetStr = "";
+    let sortStr = "";
+
+    // Fields of reference node to sort on. Since sorting is done on projection, if field is not selected
+    // sort will fail silently
+    const sortFields = ([] as string[]).concat(
+        ...(optionsInput?.sort ?? []).map((sortField) => Object.keys(sortField))
+    );
+
+    sortFields.forEach((sortField) => {
+        if (!Object.values(fieldsByTypeName[node.name]).find((r) => r.name === sortField)) {
+            fieldsByTypeName[node.name] = {
+                ...fieldsByTypeName[node.name],
+                [sortField]: { alias: sortField, args: {}, fieldsByTypeName: {}, name: sortField },
+            };
+        }
+    });
```

Comment:
This is a beautiful hack if I've ever seen one! 😅  Whilst I will admit to modifying the resolve tree in the past, it always gets me a bit on edge. But it's clearly an effective use of it here. I'm assuming there was no way of calculating the missing fields whilst generating the projection? That would likely be the preferred approach.

Reply:
You're right of course. In hindsight that seems careless and easy to avoid. Fixed in https://github.com/neo4j/graphql/pull/781/commits/e401d5ca0cc424c9204a2e731146bda8032018af

Should `createProjectionAndParams` be refactored to accept a `resolveTree` argument instead of the `fieldsByTypeName` as that would already be in `resolveTree`. It would make it easier to reason about in my opinion and one would have access to `args` as well which would greatly simplify this PR.

Reply:
A refactor is quite trivial as it turns out. Here are the necessary changes:
https://github.com/dmoree/graphql/compare/fix/sort-on-missing-fields...dmoree:refactor/create-projection-and-params

This seems clearer to me for some reason as it is evident that what you are really passing into `createProjectionAndParams` is _a_ resolve tree and when it needs to be modified from above it is quite apparent.  It did in fact simplify this PR as the modification of the fields now resides within `createProjectionAndParams` and not above it in `translateRead` or below in the `relationField` of the reducer. This is because `sort` is simply on the `args` of a resolve tree to be projected on a node.

Reply:
Interesting! A `ResolveTree` object certainly feels more sensible to pass around than `FieldsByTypeName`, which feels more like an "internal" type to the `parse-graphql-resolve-info` library. This actually looks like a really sound change, my only interest is https://github.com/dmoree/graphql/compare/fix/sort-on-missing-fields...dmoree:refactor/create-projection-and-params#diff-a9bb71892975416036eb25742d83ed7e1bbf074bf21ff16db77d5f67e98008f2R125 - what makes this call different that `node` needs to be spread into the `resolveTree`?

I would say merge that branch into this so that we can review in full. 🙂

Reply:
That link is broken on my end, but I think you are talking about in `createConnectionAndParams`? As this was simply a refactor of `createProjectionAndParams`, no variable names were changed in this file.

`node` here was defined as a `ResolveTree` coming from the edge of the connection. I would probably rename it, but I just left everything as is for the time being.

Reply:
Brilliant, thanks for the explanation - I like this refactor!